### PR TITLE
Build icons using config

### PR DESCRIPTION
**Pull request by viktor-yakubiv**
I am trying to write a script that would do exactly the same what we have in the `package.json`:

```
svgo -f src -o lib -r --disable=removeViewBox --enable=removeDimensions",
svgstore lib/assets/icons/* | svgo -i - -o lib/assets/icons.svg --disable=removeUselessDefs,cleanupIDs
```

SVGO is a quite tricky thing to manage it running from the node script ~. I am struggling to do optimisations. Apart from that~, everything works.

### Roadmap

- [x] Create icons build script
- [x] Read options from the config
- [x] Add cosmiconfig
- [ ] Add yargs or coa

### CODE REVIEW

File: bin/icons/index.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    removeUselessDefs: false,
+    cleanupIDs: false,
+  },
+}
+
+const run = async () => {
+  const config = await readConfig()
+  const paths = await resolveInput(config.input)
+  const files = await read(paths)
+  const icons = await optimize(files, sourceOptions)
```
Comment (viktor-yakubiv) [inline bin/icons/index.js:77]: For some reason, SVGO removes all content leaving only `<svg>` tag. I have no idea what the problem may happen.

File: config.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  'check-circle',
+  // 'cite',
+  'close',
+  'cog',
+  // 'core-symbol',
+  // 'dashboard',
+  // 'data',
+  'file-alert', // 'document-alert',
+  'file-check', // 'document-success',
+  'file', // 'document',
```
Comment (viktor-yakubiv) [inline config.js:16]: @Joozty be aware this is a breaking change. `core-symbol` should not be an icon. The rest will be removed or renamed according to the original icon names from `@mdi/svg` what you may see in the current file. I am working now on finding an proper replacement for everything.

In the future (maybe this, maybe next PR unless we discuss it) design will not ship any icon. The dependent project should build own sprite using provided tools.

**Comment by viktor-yakubiv** [comment]
My working directory after some testing 😄

a lot of svg directorie

File: config.js (hunk 2 above)
Comment (viktor-yakubiv) [reply config.js]: I added full mapping.

**Comment by viktor-yakubiv** [comment]
I will not be adding `yargs` or `coa`. It's not necessary at the moment and current result suffices me. @Joozty could you please review?

I will remove current icons after and remove the default config in a separate PR.

File: bin/icons/config.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+const read = async (filepath) => {
+  const explorer = cosmiconfig('design')
+  if (filepath != null) return explorer.load(filepath)
+
+  let config = await explorer.search()
+  if (config == null) config = await explorer.load(defaultFilepath)
+
+  return config
+}
```
Comment (Joozty) [inline bin/icons/config.js:39]:
```suggestion
const read = async (filepath = defaultFilepath) => {
  const explorer = cosmiconfig('design')
  return explorer.load(filepath)
}
```

File: bin/icons/config.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  const output = prepareOutput(config.output)
+  const svgo = { ...defaultConfig.svgo, ...config.svgo }
+  return { icons, output, svgo }
+}
+
+const load = async (filepath) => {
+  const { config: userConfig } = await read(filepath)
+  const config = prepare(userConfig)
+  return config
+}
```
Comment (Joozty) [inline bin/icons/config.js:99]:
```suggestion
const load = async (filepath) => {
  const { config: userConfig } = await read(filepath)
  return prepare(userConfig)
}
```

Comment (viktor-yakubiv) [reply bin/icons/config.js:99]: I wanted to leave this as it is. It allows putting `console.log` in between without changing anything.

File: bin/icons/config.js (hunk 3 above)
Comment (viktor-yakubiv) [reply bin/icons/config.js:39]: It's not like that. I want to load the default `design` config only if there is not project-specific one.

**Comment by viktor-yakubiv** [comment]
Thank you! I used a lot of type checks to make the configuration as flexible as possible.

**Pull request merged**